src/main/java/bio/terra/service/snapshotbuilder/SnapshotBuilderService.java

```diff
@@ -131,21 +139,57 @@ public SnapshotBuilderGetConceptsResponse getConceptChildren(
     return new SnapshotBuilderGetConceptsResponse().result(concepts);
   }
 
-  private int getRollupCount(UUID datasetId, List<SnapshotBuilderCohort> cohorts) {
-    return 100;
-  }
-
   public SnapshotBuilderCountResponse getCountResponse(
-      UUID id, List<SnapshotBuilderCohort> cohorts) {
+      UUID id, List<SnapshotBuilderCohort> cohorts, AuthenticatedUserRequest userRequest) {
     return new SnapshotBuilderCountResponse()
         .sql("")
-        .result(new SnapshotBuilderCountResponseResult().total(getRollupCount(id, cohorts)));
+        .result(
+            new SnapshotBuilderCountResponseResult()
+                .total(
+                    getRollupCountForCriteriaGroups(
+                        id,
+                        cohorts.stream().map(SnapshotBuilderCohort::getCriteriaGroups).toList(),
+                        userRequest)));
   }
 
   public EnumerateSnapshotAccessRequest enumerateByDatasetId(UUID id) {
     return convertToEnumerateModel(snapshotRequestDao.enumerateByDatasetId(id));
   }
 
+  public int getRollupCountForCriteriaGroups(
+      UUID datasetId,
+      List<List<SnapshotBuilderCriteriaGroup>> criteriaGroupsList,
```

Comment:
Just a small nit, personally I wouldn't include `list` in the name as list is a type; it could be a collection or other type of group
```suggestion
      List<List<SnapshotBuilderCriteriaGroup>> criteriaGroups,
```